src/frontends/pytorch/src/transforms/aten_getitem_replacer.cpp

```diff
@@ -0,0 +1,104 @@
+                auto split_size_1d = std::make_shared<opset8::Unsqueeze>(split_size, const_0);
+                auto axis = torch_split->get_input_source_output(2);
+                auto axis_1d = std::make_shared<opset8::Unsqueeze>(axis, const_0);
+                auto getitem_idx = getitem->input(1).get_source_output();
+
+                // Calculate number of splits based on input shape and split_size.
+                auto shape = std::make_shared<opset8::ShapeOf>(input);
+                auto len_to_split = std::make_shared<opset8::Gather>(shape, axis, const_0);
+                // Convert to f64 from int to calculate reminder - last chunk can be smaller if Shape in given axis is
+                // not equally divisible.
+                auto len_to_split_float = std::make_shared<opset8::Convert>(len_to_split, element::f64);
+                auto split_size_1d_float = std::make_shared<opset8::Convert>(split_size_1d, element::f64);
+                auto out_div = std::make_shared<opset8::Divide>(len_to_split_float, split_size_1d_float);
+                auto out_num = std::make_shared<opset8::Ceiling>(out_div);
+                auto out_num_0d = std::make_shared<opset8::Squeeze>(out_num, const_0);
+
+                // Use Range and Gather to convert negative getitem indexes into positive due problems with indexing
+                // with -1.
+                auto possible_out_idx =
+                    std::make_shared<opset8::Range>(const_0_0d, out_num_0d, const_1_0d, split_size.get_element_type());
+                auto always_positive_out_idx = std::make_shared<opset8::Gather>(possible_out_idx, getitem_idx, const_0);
+
+                // Use Slice to get only split output selected by getitem idx. Couldn't use VariadicSplit due to
+                // problems with dynamic inputs.
+                auto split_slice_start = std::make_shared<opset8::Multiply>(always_positive_out_idx, split_size_1d);
+                auto split_slice_end = std::make_shared<opset8::Add>(split_slice_start, split_size_1d);
+                auto split =
+                    std::make_shared<opset8::Slice>(input, split_slice_start, split_slice_end, const_1, axis_1d);
+                copy_runtime_info({getitem, input_node}, split);
+                replace_node(getitem, split);
+            } else {
+                auto split = std::make_shared<opset8::VariadicSplit>(torch_split->get_input_source_output(0),
+                                                                     torch_split->get_input_source_output(2),
+                                                                     torch_split->get_input_source_output(1));
+                OutputVector res;
+                auto index = 0;
+                if (index_val[0] >= 0) {
+                    index = index_val[0];
+                } else {
+                    index = split->outputs().size() + index_val[0];
+                }
+                res.push_back(split->outputs()[index]);
```

Comment:
looks like you can create `res` here instead of calling `push_back`.

Reply:
Done